### ISSUE TITLE
Cannot provide my own sampling strategy

### ISSUE THREAD

**Issue by richardgavel** [aws/aws-xray-sdk-dotnet#18]
I am trying to write my own sampling strategy, however, the AWSXRayRecorder.Instance has the LocalizedSamplingStrategy instead of mine. From looking at the code, it seems like the InitializeInstance method I am using builds a new AWSXRayRecorder based on instance I am providing, but the sampling strategy does not carry over.

```
AWSXRayRecorder.InitializeInstance(Configuration, BuildAWSXRayRecorder());

private static AWSXRayRecorder BuildAWSXRayRecorder()
        {
            var builder = new AWSXRayRecorderBuilder()
                .WithSamplingStrategy(new AWSXRaySamplingStrategy());

            var result = builder.Build();

            return result;
        }
```

**Comment by yogiraj07** [comment]
Hi @richardgavel ,

Thank you for your post.

`InitializeInstance` method populates the provided recorder with the configuration. However, this method doesn't set the provided recorder to `AWSXRayRecorder.Instance`. Hence, whenever you use AWSXRayRecorder.Instance in the later part of your code, a default new instance is created, which overrides the sampling strategy . 
For reference : https://github.com/aws/aws-xray-sdk-dotnet/blob/master/sdk/src/Core/AWSXRayRecorder.netcore.cs#L81
For your use case, please use the `recorder` instance you created, to create segment/subsegment.

I will take this issue on priority and will set the custom recorder to `AWSXRayRecorder.Instance`. 

Thanks,
Yogi

**Comment by yogiraj07** [comment]
Hi @richardgavel 

We have included this fix in `2.2.0-beta` release.

Please refer release details : https://github.com/aws/aws-xray-sdk-dotnet/blob/master/CHANGELOG.md

Feel free to open this issue if you have any questions.

Thanks,
Yogi
